python/src/trezorlib/ui.py

```diff
@@ -71,6 +78,11 @@ def echo(*args: Any, **kwargs: Any) -> None:
 
 
 def prompt(*args: Any, **kwargs: Any) -> Any:
+    # Disallowing hidden input when it would cause issues
+    if not CAN_HANDLE_HIDDEN_INPUT:
+        if "hide_input" in kwargs and kwargs["hide_input"]:
+            kwargs["hide_input"] = False
+            print("(Unfortunately this terminal cannot handle hidden input.)")
```

Comment:
1. `click.echo` is probably what we want instead of `print`
2. let's try a slightly different wording

```suggestion
            click.echo("(Warning: This terminal can't handle hidden input. The input may be shown!)")
```

Reply:
After the change, the output looks like this. Is it alright?

![image](https://user-images.githubusercontent.com/42543243/144038227-00e2d164-4f51-4b7e-96b5-8ee2f1da3182.png)

Reply:
I am fine with the outcome, but let you and @matejcik have the final word.

Reply:
not "input _may_ be shown", we know it _will_ be shown.

As an alternative to the `click.echo` call, maybe modify the `text`, e.g.: `text += " (WARNING: will be displayed!)`
But I'm not super happy about the wording just yet.

Reply:
also even if we decide to go with echo, it must be just `echo`, not `click.echo` -- because `echo` goes to stderr while `click.echo` without `err=True` goes to stdin.